### PR TITLE
feat: Make hostname for status badges configurable (#8024)

### LINKED ISSUE #8024: Make hostname for status badges configurable
# Summary

Make hostname for status badges configurable.
This only matters for the Add Details UI, where the user gets the URL for the status badge.

# Motivation

Our instance is on an internal IP, but I have a separate public endpoint that is filtered and allows particular requests to get through.
For status badges to work when you are not on our network, users must use the public endpoint, which is a different URL.

# Proposal

How do you think this should be implemented?

Add a config option that allows you to specify the URL for status badges. This must be separate from the root URL, as changing that will also affect SSO.

### PR DESCRIPTION
**Pull request by keithchong**
Signed-off-by: Keith Chong <kykchong@redhat.com>

Fixes https://github.com/argoproj/argo-cd/issues/8024

This provides the option to replace the hostname part of the URL as requested by the originator of the issue.

I have not updated the status badge pages in the operator manual or user guide.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: docs/operator-manual/argocd-cm.yaml [hunk 1]
```diff
@@ -13,6 +13,12 @@ data:
   # Enables application status badge feature
   statusbadge.enabled: "true"
 
+  # Override the Argo CD hostname root URL for both the project and the application status badges.
+  # Here is an example of the application status badge for the app `myapp` to see what is replaced.
+  #    <statusbadge.url>api/badge?name=myapp&revision=true
+  # Uncomment and provide custom URL to override. You must include the trailing forward slash:
+  #statusbadge.url: "https://cd-status.apps.argoproj.io/"
```
Comment (jannfis) [inline docs/operator-manual/argocd-cm.yaml:20]:
```suggestion
  # Provide custom URL to override. You must include the trailing forward slash:
  statusbadge.url: "https://cd-status.apps.argoproj.io/"
```

File: ui/src/app/shared/components/badge-panel/badge-panel.tsx [hunk 2]
```diff
@@ -9,66 +9,67 @@ require('./badge-panel.scss');
...
-        badgeURL = `${root}api/badge?name=${app}&revision=true`;
-        entityURL = `${root}applications/${app}`;
-        alt = 'App Status';
-    } else if (project) {
-        badgeURL = `${root}api/badge?project=${project}&revision=true`;
-        entityURL = `${root}projects/${project}`;
-        alt = 'Project Status';
-    } else {
+    if (!app && !project) {
         throw new Error('Either app of project property must be specified');
```
Comment (jannfis) [inline ui/src/app/shared/components/badge-panel/badge-panel.tsx:13]: I know this is not part of your change, but I just spotted this little typo.

```suggestion
        throw new Error('Either app or project property must be specified');
```

**Review by jannfis — COMMENTED** [review]
Just a few nitpicks, otherwise LGTM. 

Can't comment on the UI code tho :)

**Review by crenshaw-dev — APPROVED** [review]
Tested locally, LGTM

**Review by alexmt — APPROVED** [review]
Thank you!

File: docs/operator-manual/argocd-cm.yaml (hunk 1 above)
Comment (keithchong) [reply docs/operator-manual/argocd-cm.yaml]: Ok, merged.

**Comment by keithchong** [comment]
Thanks everyone for reviewing this.  Suggested changes have been merged.

**Pull request merged**